pinot-core/src/main/java/org/apache/pinot/core/query/pruner/ColumnValueSegmentPruner.java

```diff
@@ -59,8 +62,11 @@
 @SuppressWarnings({"rawtypes", "unchecked"})
 public class ColumnValueSegmentPruner implements SegmentPruner {
 
+  private int _inPredicateThreshold;
+
   @Override
   public void init(PinotConfiguration config) {
+    _inPredicateThreshold = config.getProperty(Server.CONFIG_OF_VALUE_PRUNER_IN_PREDICATE_THRESHOLD, Server.DEFAULT_VALUE_PRUNER_IN_PREDICATE_THRESHOLD);
```

Comment:
This is incorrect because when the config is passed from the `SegmentPrunerProvider`, the prefix is already removed, and here you should use key `inpredicate.threshold` to access the config.
The reason why the test works is because you directly pass the top-level config into this pruner class, which is not the case in the production code.

Reply:
done!